### PR TITLE
[#172885142] block blacklisted users

### PR DESCRIPTION
**Pull request by AleDore**
added user login block check. If a user has requested to delete its own data, login is blocked until all data are processed, in this time range login returns 403.

### CODE REVIEW

File: src/services/redisSessionStorage.ts [hunk 1]
```diff
@@ -307,6 +308,14 @@ export default class RedisSessionStorage extends RedisStorageUtils
       .run();
   }
 
+  public async userHasLoginBlocked(
+    fiscalCode: FiscalCode
+  ): Promise<Either<Error, boolean>> {
+    return fromEither(
+      tryCatch(() => this.redisClient.exists(`USER-BLOCKED-${fiscalCode}`))
```
Comment (gunzip) [inline src/services/redisSessionStorage.ts:315]: @cloudify we have to implement a way to temporarely block users while the data deletion procedure is running.

we've chosen to put a flag 'block-user' into redis for each blocked CF. we can implement it either:
1) with a USER-BLOCKED-<CF> entry
2) with a USER-BLOCKED redis Set

what do you think about this approach? 
(2 will let us to enumerate blocked users without a full scan on redis keys)

File: src/services/redisSessionStorage.ts [hunk 2]
```diff
@@ -307,6 +308,14 @@ export default class RedisSessionStorage extends RedisStorageUtils
       .run();
   }
 
+  public async userHasLoginBlocked(
+    fiscalCode: FiscalCode
+  ): Promise<Either<Error, boolean>> {
```
Comment (BurnedMarshal) [inline src/services/redisSessionStorage.ts:313]: `exists` method returns an integer.
From the SDK tests [here](https://github.com/NodeRedis/node-redis/blob/79927c56a0e2fb7819afbb0e13c6078d0015564e/test/commands/exists.spec.js) and from the spec [here](https://redis.io/commands/exists)

Comment (AleDore) [reply src/services/redisSessionStorage.ts:313]: so there are errors on typos?

File: src/services/redisSessionStorage.ts (hunk 1 above)
Comment (BurnedMarshal) [reply src/services/redisSessionStorage.ts]: @gunzip `EXISTS` ([here](https://redis.io/commands/exists)) and `HGET` ([here](https://redis.io/commands/hget)) have the same complexity `O(1)`

Comment (gunzip) [reply src/services/redisSessionStorage.ts]: yes it's the enumeration complexity that changes (not the lookup). if you want to list all blocked users without a set you have to do a full scan of all redis keys (same if you want to clear the list of blocked users)

**Pull request merged**